### ISSUE TITLE
Unable to swipe through pager on a button

### ISSUE THREAD

**Issue by teameh** [fermoya/SwiftUIPager#55]
When adding a button to a page it's not possible to swipe the pager on the button any more. 

Is there a way to make this work? 

Here's an example where I've [added a button](https://github.com/teameh/SwiftUIPager/commit/4530061f8e4723b1d8f521246b800fcbec844076) around the text: 

```
    func pageView(_ page: Int) -> some View {
        ZStack {
            Rectangle()
                .fill(Color.yellow)
            Button(action: { print("Button tapped") }) {
                Text("Page: \(page)")
                    .bold()
            }
            .buttonStyle(PlainButtonStyle())
        }
        .cornerRadius(5)
        .shadow(radius: 5)
    }
```

![pager-swipe](https://user-images.githubusercontent.com/1330668/83866026-719e8a00-a727-11ea-85d1-1540248614b5.gif)

**Comment by fermoya** [comment]
Unfortunately there isn't. Currently in `SwiftUI` it is possible to set priorities but not make a gesture fail. The `Button` should fail the recognition of the gesture and this way delegate its touch.

* If you try to set `swipeGesture` in `Pager` as `simultaneousGesture`, then both gestures are recognized
* Change it to `highPriorityGesture` and then you can't use any gestures inside the `Pager`.

I created a modifier `highPriorityGesture()` to modify the `minimumDistance` required for `DragGesture` to be recognized, but I believe it won't work because it targeted the scenario where `Pager` was embedded into an interactive modal. 

See issues #48 and #18 as well as [PR-41](https://github.com/fermoya/SwiftUIPager/pull/41) for more info

**Comment by teameh** [comment]
Hmm, that's tough, thanks for linking those issue's, I'll take a look.

**Comment by teameh** [comment]
> I created a modifier highPriorityGesture() to modify the minimumDistance required for DragGesture to be recognized, but I believe it won't work because it targeted the scenario where Pager was embedded into an interactive modal.

That indeed does not work. 

> If you try to set `swipeGesture` in `Pager` as `simultaneousGesture`, then both gestures are recognized

Changing that makes it work fine! What would be the downside of doing that? Why are you not using `simultaneousGesture`?

**Comment by fermoya** [comment]
So the problem is I use `simultaneousGesture` both the `Button` and `DragGesture` action callbacks would get called.

So if you take a look at issue #48, the problem is similar. I shared there a link to a SO question, find it [here](https://stackoverflow.com/questions/58440469/swiftui-navigationlink-and-scrollview-drag-gesture-colliding). 

So both behaviors are wrong (the current behavior and the one described on SO). I just wonder which one “feels better”. I have `minimumDistance` to “delay” the recognition of the `DragGesture`. This makes it work inside a ScrollView. Right now I don’t remember but there might be something similar in`TapGesture` (you’d have to use instead of `Button`).

To be honest, this situation with the gesture is driving me insane, I don’t manage to find the right way to do this, in case there is any at all at the moment. It’s a bummer because gestures are so important in a mobile device.

**Comment by fermoya** [comment]
Oh, i just remembered. If `simulatenousGesture` is used, then you can’t nest `Pager` either. We’d need modifiers to set priorities or basically decide which gesture to use. It gets very messy to be honest.

**Comment by teameh** [comment]
Check. Yeah it does get a bit messy. Not an easy tradeoff. 

You could argue though that adding and having to use a modifier to be able to interact with stuff like buttons is less of an evil then not be able to use it at all. Same for the nesting, it's a bit hard to explain but having to need a modifier to be able to nest it sounds not that awful.

Have you heard of a lot of users nesting them?

I haven't really dived into gestures in SwiftUI yet, too bad Apple did not make them a top prio in SwiftUI

**Comment by fermoya** [comment]
Some people mentioned nesting `Pager` before. It's just not well designed, there's something missing here because gestures get recognized even when they shouldn't. Like see #58. A `MagnificationGesture` shouldn't collide with `DragGesture` as the first needs both fingers whereas the second just needs one. It's all very strange

**Comment by teameh** [comment]
Hm, yeah that is strange. Hopefully WWDC will bring some changes to the gesture system then. 

For now I'm just adding here that this worked for me instead of using a button: 

```
func pageView(_ page: Int) -> some View {
        ZStack {
            Rectangle()
                .fill(Color.yellow)
            Text("Page: \(page)")
                .bold()
                .simultaneousGesture(
                    TapGesture().onEnded { print("Button tapped") }
                )
                .accessibility(label: Text("Button"))
        }
        .cornerRadius(5)
        .shadow(radius: 5)
    }
```

**Comment by fermoya** [comment]
Nothing new related to gestures as far as I can see but the issue where both gestures get called seem to be solved in SwiftUI 2.0. I've created a new modifier [`pagingPriority`](https://github.com/fermoya/SwiftUIPager/blob/a0cab6693e2fe464f3c5871b877f5efe6313fd71/Sources/SwiftUIPager/Pager%2BBuildable.swift#L45) to choose the best way to embed [`swipeGesture`](https://github.com/fermoya/SwiftUIPager/blob/a0cab6693e2fe464f3c5871b877f5efe6313fd71/Sources/SwiftUIPager/Pager.swift#L211). If you use `pagingPriority(.simultaneous)` you'll be able to swipe buttons.

See [_v1.7.0_](https://github.com/fermoya/SwiftUIPager/releases/tag/1.7.0) for more info.
